Makefile

```diff
@@ -49,13 +49,21 @@ update:
 	mkdir -p .cargo
 	cp $< $@
 
-vendor: .cargo/config
+vendor.tar.xz:
 	cargo vendor
-	touch vendor
+	tar pcfJ vendor.tar.xz vendor
+	rm -rf vendor
+
+vendor: .cargo/config vendor.tar.xz
+
+tests:
+	cargo test
 
 $(BINARY): $(SRC)
-	if [ -d vendor ]; \
+	if [ -f vendor.tar.xz ]; \
 	then \
+		tar pxf vendor.tar.xz; \
+		rm vendor.tar.xz; \
```

Comment:
Is there a reason to remove the vendor tarball here?

Reply:
No real reason. it's removed automatically on the next build.